### ISSUE TITLE
Autocrafting book recipes moving crafting slot items

### ISSUE THREAD

**Issue by emerickchasse** [RockinChaos/ItemJoin#614]
### Server Version

PaperSpigot 1.20.1

### Resource Version

 ItemJoin v6.1.0

### Dump Information

https://ci.craftationgaming.com/dump/?id=meqosigeja

### Describe The Bug

Autocrafting book, recipes move crafting slot items that are not interatable causing them to shift on hot bar and cant be moved.

### Expected Behavior

Not moving the items when using autocraft book (even recipe with items I don't have)

### Steps To Reproduce

1. Set items to crafting slots
2. Open Autocrafting Recipe book
3. Choose an item you cant craft
4. Items will shift into hotbar, even if they are not moveable
![image](https://github.com/user-attachments/assets/3839357d-6083-4473-bc2d-b93027825583)

### Checklist

- [X] I am running latest version of this resource.
- [X] I have read the wiki to make sure it's not an issue with configuration.
- [ ] I ticked all of these boxes without actually reading them.
- [X] I have checked that my bug report is unique and a previous report does not exist.

**Comment by RockinChaos** [comment]
Thanks for providing your dump url, you said 1.20.1 but it looks like your running 1.21.1. In 1.20.5 they changed how the recipe packets work so I implemented a fix back then, but apparently... in 1.21+ they changed it back for whatever reason so this has been broken for some time.

Thanks for reporting the issue! I made some changes and it appears to be functional again in the latest snapshot.
Link; https://ci.craftationgaming.com/downloads
**Remember to remove your old ItemJoin.jar file**

Let me know!~

**Comment by emerickchasse** [comment]
> **Remember to remove your old ItemJoin.jar file**
Yeah you right I am on Paper 1.21.1 sorry. I upgraded to dev build 1043 (v6.1.1-SNAPSHOT-b1043) and still experimenting the same issue.
![image](https://github.com/user-attachments/assets/f3bc5026-8324-49e9-8fbc-59fa1bd2844d)

Here is my new dump: https://ci.craftationgaming.com/dump/?id=ibuwerisok

**Comment by RockinChaos** [comment]
> > **Remember to remove your old ItemJoin.jar file**
> > Yeah you right I am on Paper 1.21.1 sorry. I upgraded to dev build 1043 (v6.1.1-SNAPSHOT-b1043) and still experimenting the same issue.
> > ![image](https://private-user-images.githubusercontent.com/133250532/386847586-f3bc5026-8324-49e9-8fbc-59fa1bd2844d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE4MTgxMTksIm5iZiI6MTczMTgxNzgxOSwicGF0aCI6Ii8xMzMyNTA1MzIvMzg2ODQ3NTg2LWYzYmM1MDI2LTgzMjQtNDllOS04ZmJjLTU5ZmExYmQyODQ0ZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTExN1QwNDMwMTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjFlNzk0ZDRhMDg3NTFkMDc4MWJkMWI1NzdlY2JlMjFkYmEyNWIyYzQzMTllNjVhYmUyOGZjYTJhODM2YmE3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.BVSKlE7dQ6kJeUeMs57dhW6v77kIHc63SQVaWwbkSK4)
> 
> Here is my new dump: https://ci.craftationgaming.com/dump/?id=ibuwerisok
> 
> I tested as OP and non-OP and it's the same thing.

Well, that's rather disappointing. Seems paper decided to change their server-side packet names so it broke the naming convention across other server forks. I was primarily testing with spigot so I didn't encounter this issue.

I made some changes specific to paper that seem to have resolved the issue from my testing. Sorry about that.
Link; https://ci.craftationgaming.com/downloads
**Remember to remove your old ItemJoin.jar file**

Let me know!~

**Comment by emerickchasse** [comment]
Yes, it's fixed !! Thank you
